addons/mail/static/src/core/messaging_service.js

```diff
@@ -205,6 +205,45 @@ export class Messaging {
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this.threadService.insert({
+                            id: notif.payload.id,
+                            model: "discuss.channel",
+                        });
+                        const filteredStarredMessages = [];
+                        let starredCounter = 0;
+                        for (const msg of this.store.discuss.starred.messages) {
+                            if (msg.resModel !== thread.model || msg.resId !== thread.id) {
+                                filteredStarredMessages.push(msg);
+                            } else {
+                                starredCounter++;
+                            }
+                        }
+                        this.store.discuss.starred.messages = filteredStarredMessages;
+                        this.store.discuss.starred.counter -= starredCounter;
+                        this.store.discuss.inbox.messages =
+                            this.store.discuss.inbox.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
+                        this.store.discuss.inbox.counter -= thread.message_needaction_counter;
```

Comment:
I'm not sure exactly how, but I found a case where this was incorrect, it decremented more than necessary.
Maybe we should count the messages that we remove, similar than in starred box?

Reply:
I am not sure if that would work either. Because it might be possible that the neeadAction messages haven't been fetched yet but they do exist on the server. So, the counter won't be decremented at all. Which will at the end require to reload.

Reply:
It seems that the `message_needaction_counter` is broken.

Reply:
For `message_needaction_counter` it's fine to ignore the issue for now.